### ISSUE TITLE
485507D2 - 天外魔境ＺＩＲＩＡ

### ISSUE THREAD

**Issue by GoldenPhilemon** [xenia-canary/game-compatibility#36]
[Marketplace](https://marketplace.xbox.com/ja-JP/Product/%E5%A4%A9%E5%A4%96%E9%AD%94%E5%A2%83%EF%BC%BA%EF%BC%A9%EF%BC%B2%EF%BC%A9%EF%BC%A1/66acd000-77fe-1000-9115-d802485507d2)

Tested on https://github.com/xenia-canary/xenia-canary/commit/660559bd372e34f62979b43fc9bf23b81f236037

# Issues:
Issues upon loading. Just hangs at a white screen with the following log issue.
i> 00000028 RtlNtStatusToDosError C000000D => 57

# Log:
[xenia.zip](https://github.com/xenia-canary/game-compatibility/files/4505849/xenia.zip)

# Screenshot(s):
![](https://user-images.githubusercontent.com/46737136/79794727-9e2d4c80-8307-11ea-8dff-0093880317d4.JPG)

# Labels:
state-hang

**Comment by Sowa95** [comment]
Tested on: https://github.com/xenia-project/xenia/commit/7003e54e510fa2169a9434cdb8d608bff7f7be1e

[xenia.zip](https://github.com/user-attachments/files/17176699/xenia.zip)

Also known as: Tengai Makyou: Ziria – Haruka naru Jipang, an X360 exclusive remake of the original game Tengai Makyou: Ziria.
No issues at first glance.

![1](https://github.com/user-attachments/assets/e710fb46-dca8-4105-a9d2-c829924e7840)
<details><summary>More screenshots</summary>
<p>

![2](https://github.com/user-attachments/assets/dbe4d1eb-3d34-4049-ae14-06baecb86f0f)
![3](https://github.com/user-attachments/assets/1adf7e14-c9f7-4261-83b8-06f16bea6201)

</p>
</details>

backwards-compatibility-unsupported | state-gameplay

**Comment by GoldenPhilemon** [comment]
Thanks for the feedback!

Yeah, the game will load and play normally up until you enter a battle.

![485507D2 - 2024-09-28T15-49-20](https://github.com/user-attachments/assets/a47fc4d0-e3c5-4281-a0d8-65bf243a6ce2)

When you enter a battle:

![485507D2 - 2024-09-28T15-49-43](https://github.com/user-attachments/assets/bb216469-9007-4963-a433-d8e9d390a890)

And then you attack an enemy:

![485507D2 - 2024-09-28T15-49-58](https://github.com/user-attachments/assets/77cbffcb-b0bf-4168-ae5d-42ace913bffb)

As soon as you attack them, the main character will disappear and then the game will just hang while the music plays.

![485507D2 - 2024-09-28T15-50-05](https://github.com/user-attachments/assets/f47624cc-bd1c-48fb-a557-2d33d8552c2f)


So unfortunately the game won't be playable until that's fixed where your character won't disappear when attacking in battles haha

This was tested on canary_experimental@952526a78.

Let me know if you have any questions though.

Thank you so much!

**Comment by RanceJustice** [comment]
Last I checked the issue is still ongoing as described by GoldenPhilemon; everything seems to go smoothly until a player enters battle where combat will hang.   There doesn't seem to be any  more recent emulated gameplay  visible on YouTube showing examples to the contrary, either.

**Comment by GoldenPhilemon** [comment]
Yeah, the issue is still occurring for me as well in the latest Xenia/latest nightly build.

Hopefully the issue can be fixed soon :)

Thank you so much!

**Comment by Left-Empty** [comment]
This last week of canary updates brought some slight changes: now sometimes hits can be exchanged. Eventually the game locks, either after a turn or at the end of the fight. Occasionally, Ziria does not disappear anymore too.

**Comment by GoldenPhilemon** [comment]
Yeah, the issue is still occurring for me as well via the nightly Xenia Canary builds unfortunately.

Hopefully it gets resolved/fixed soon :)

**Comment by nero81af** [comment]
I started debugging this, but it's really hard: probably there's a missing kernel call that should signal something after I get hit or I finish the battle

**Comment by nero81af** [comment]
here a small recap of my findings trying to go through disassembly without luck

Findings:
  - Bug = guest game spin-polls a memory address (~`0x82EBE384`) that xenia never updates. Not a kernel wait.
  - Hot function: 82646A88 (timer compute), called from FSM at 82155C28.
  - Music + render keep running because they're on other threads; only the game-logic thread is stuck.

  Tried (none fixed it):
  - ~13 cvar combos (clock, audio, GPU backend, threading, MMIO, APC, time scalar)
  - Source patches (APC shotgun, VdCallGraphicsNotificationRoutines stub fill, wait-drain APCs)
  - HIR dump for 10672 functions

  Blocker: xenia's fmt v12 crashes on most disasm formatting → can't dump PPC instructions to find polled address.

**Comment by Left-Empty** [comment]
Fights seem to work in the latest Xenia-edge version! (d39b10a)

There are some slight line artefacts in some models that weren't present before (easily noticeable on Ziria's own hairs). Those were easily fixed by setting precise_interpolation to false on my computer (no upscaling on a Core i5, Nvidia 2070RTX, Win10 and D3D12)

**Comment by GoldenPhilemon** [comment]
> Fights seem to work in the latest Xenia-edge version! (d39b10a)

Yes! I just tested it and it works without freezing now!

I can finally play the game normally! This is an amazing development!

Thank you so much to all the people that made this possible :)

**Comment by Left-Empty** [comment]
I am about 25 hours in, and had experienced zero issue thus far, until a small one today.
When entering the Gotemba town for the first time by the Mount Fuji, Hoteimaru got a little flashback events, and it is really dark. You can make some shapes if you squint, and it seems the image is surrounded by a halo, which might be the cause of the whole rendering issue. I tried in accuracy mode, with the same result.

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/ff33ee15-f295-4198-8df9-e64e16e32999" />

Here is a save file (I had to replay quite a big chunk of the game as it was a bit greedy on the save side and had to do something unintuitive to save right before the issue ;)

[SaveData_02.zip](https://github.com/user-attachments/files/31971328/SaveData_02.zip)

And finally, if you want to experience the issue first-hand, here is a map with your location and where you are to go.

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/1c5b5166-434e-4000-95f1-599ca9aa3a63" />

Go to the marked dot, a wooden area: you can go through it simply by walking on the paved way — you shouldn't encounter enemies that way. When you exit the area, the town is right outside, and on entering it you will trigger the event.

**Comment by Left-Empty** [comment]
I played the whole game on stream this past month and I am happy to say it is totally completable, probably 100% so. I played the main game and spent quite a few hours doing silly side stuff (that served no real purpose, as the game gives you everything you need on the default path), and I experienced the one single glitch mentionned above in my 50 hours playthrough. I kept updating Xenia-edge daily during this and not a single issue on that front either.
I vouch for a state-playable!

**Comment by GoldenPhilemon** [comment]
That's really great to hear that it sounds like the game is fully playable except for that one flashback graphical issue :)

Feel free to keep us posted if any new Xenia Edge updates ever fix the issue mentioned above.

Thank you again for all of your help!

**Comment by GoldenPhilemon** [comment]
Just to quickly ask, do we know if any builds of Xenia Canary also fix the gameplay issue?

Or is it only fixed on Xenia Edge for now?

Thank you so much for the help!
